### DISCUSSION TITLE
Why double and decimal columns changed?

### DISCUSSION THREAD

**Discussion by ThePlatinum** [laravel/framework#50714] [Q&A] [answered]
Working with Laravel 11, I noticed a change in the column types for double and decimal. I am wondering why and I would love if someone can explain why the new way is better:

Here are screenshots and texts from the official doc to help explain the change.

From Laravel 11, only the decimal column had the precision and scale

The decimal method creates a DECIMAL equivalent column with the given precision (total digits) and scale (decimal digits): `$table->decimal('amount', total: 8, places: 2);`

The double method creates a DOUBLE equivalent column: `$table->double('amount');`

![Screenshot from 2024-03-22 11-29-50](https://github.com/laravel/framework/assets/30179079/42d7ce33-9a65-40a8-8a99-a97984bcba6d)

From Laravel 10, both the decimal and double column had the precision and scale

The decimal method creates a DECIMAL equivalent column with the given precision (total digits) and scale (decimal digits): `$table->decimal('amount', $precision = 8, $scale = 2);`

The double method creates a DOUBLE equivalent column with the given precision (total digits) and scale (decimal digits): `$table->double('amount', 8, 2);`

![Screenshot from 2024-03-22 11-28-48](https://github.com/laravel/framework/assets/30179079/430a854a-18f9-4a6e-9630-29108d0fa1d8)

Thank you.

**Comment by hafezdivandari** [comment, marked as answer]
Decimal type has not been changed, just float and double types have been rewritten to be consistent across all databases.

You may check the explaination on the upgrade guide:
https://laravel.com/docs/11.x/upgrade#floating-point-types

And also the releared PR: https://github.com/laravel/framework/pull/48861

**Reply by ThePlatinum** [reply]
Thank you.

I will learn to read the upgrade and change log next time.
